### PR TITLE
cli overhaul

### PR DESCRIPTION
**Pull request by debris**
changes:
- fixed incorrect parsing of decimal numbers
- changed cache cli options and fixed their default values (cache will be working now)
- fixed inconsistency in rpc modules started
- removed `die!`, cause it's not testable
- added notifications about deprecated cli options
- removed unnecessary complexity in multiple places
- tests, tests, tests

- many (not all) command line params are now parsed to intermediate structures which validate them

### CODE REVIEW

File: parity/cache.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	}
+}
+
+impl CacheConfig {
+	/// Creates new cache config with cumulative size equal `total`.
+	pub fn new_with_total_cache_size(total: u32) -> Self {
+		CacheConfig {
+			rocksdb: total / 2,
+			blockchain: total / 4,
+			queue: total / 4,
```
Comment (arkpar) [inline parity/cache.rs]: Block queue is not cache. There should be a minimum of  ~4 MB at least

File: parity/cache.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		CacheConfig::new_with_total_cache_size(200)
+	}
+}
+
+impl CacheConfig {
+	/// Creates new cache config with cumulative size equal `total`.
+	pub fn new_with_total_cache_size(total: u32) -> Self {
+		CacheConfig {
+			rocksdb: total / 2,
+			blockchain: total / 4,
```
Comment (arkpar) [inline parity/cache.rs]: This should default to a few megabytes. Querying old blocks is an extremely rare case.

File: parity/cli.rs [hunk 3]
```diff
@@ -201,18 +202,16 @@ Footprint Options:
...
-  --cache-pref-size BYTES  Specify the preferred size of the blockchain cache in
-                           bytes [default: 16384].
-  --cache-max-size BYTES   Specify the maximum size of the blockchain cache in
-                           bytes [default: 262144].
-  --queue-max-size BYTES   Specify the maximum size of memory to use for block
-                           queue [default: 52428800].
-  --cache MEGABYTES        Set total amount of discretionary memory to use for
+  --cache-size-db MB       Override RocksDB database cache size [default: 100].
+  --cache-size-blocks MB   Specify the prefered size of the blockchain cache in
+                           bytes [default: 50].
```
Comment (arkpar) [inline parity/cli.rs]: 50 bytes?

File: parity/cli.rs [hunk 4]
```diff
@@ -201,18 +202,16 @@ Footprint Options:
...
                            auto - use the method most recently synced or
                            default to fast if none synced [default: auto].
-  --cache-pref-size BYTES  Specify the preferred size of the blockchain cache in
-                           bytes [default: 16384].
-  --cache-max-size BYTES   Specify the maximum size of the blockchain cache in
-                           bytes [default: 262144].
-  --queue-max-size BYTES   Specify the maximum size of memory to use for block
-                           queue [default: 52428800].
-  --cache MEGABYTES        Set total amount of discretionary memory to use for
+  --cache-size-db MB       Override RocksDB database cache size [default: 100].
```
Comment (arkpar) [inline parity/cli.rs]: No need to mention rocksdb here. Most users have no idea what that is.

**Comment by gavofyork** [comment]
is it "still in progress" as is claimed in the PR's description?

**Comment by debris** [comment]
no

File: parity/cli.rs [hunk 5]
```diff
@@ -106,7 +107,7 @@ API and Console Options:
                            interface. APIS is a comma-delimited list of API
                            name. Possible name are web3, eth, net, personal,
                            ethcore, ethcore_set, traces.
-                           [default: web3,eth,net,ethcore,personal,traces].
+                           [default: web3,eth,net,ethcore,personal,traces,rpc].
```
Comment (gavofyork) [inline parity/cli.rs]: `rpc` option not covered in above description

Comment (NikVolf) [inline parity/cli.rs]: it's a meta-api, probably no reason to configure it in the cli, it should be just silently included
imo

Comment (gavofyork) [inline parity/cli.rs]: then it shouldn't really be in the user-visible options

**Comment by gavofyork** [comment]
might be that there is a better solution than exposing `rpc` as an optional RPC.

**Pull request merged**